ui/pages/confirmations/components/confirm/info/shared/advanced-details/advanced-details.tsx

```diff
@@ -0,0 +1,174 @@
+import { TransactionMeta } from '@metamask/transaction-controller';
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  ConfirmInfoRow,
+  ConfirmInfoRowText,
+} from '../../../../../../../components/app/confirm/info/row';
+import {
+  Box,
+  Button,
+  ButtonSize,
+  ButtonVariant,
+  Icon,
+  IconName,
+} from '../../../../../../../components/component-library';
+import Tooltip from '../../../../../../../components/ui/tooltip';
+import {
+  BlockSize,
+  BorderColor,
+  IconColor,
+} from '../../../../../../../helpers/constants/design-system';
+import { useCopyToClipboard } from '../../../../../../../hooks/useCopyToClipboard';
+import { useI18nContext } from '../../../../../../../hooks/useI18nContext';
+import {
+  currentConfirmationSelector,
+  getCustomNonceValue,
+  getNextSuggestedNonce,
+  getUseNonceField,
+} from '../../../../../../../selectors';
+import {
+  getNextNonce,
+  showModal,
+  updateCustomNonce,
+} from '../../../../../../../store/actions';
+import { useKnownMethodDataInTransaction } from '../../hooks/known-method-data-in-transaction';
+
+const NonceDetails = () => {
+  const t = useI18nContext();
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(getNextNonce());
+  }, [dispatch]);
+
+  const useNonceField = useSelector(getUseNonceField);
```

Comment:
Is this the preference to allow custom nonces? In which case should we be more explicit and call it `enableCustomNonce`?